src/wallet-account.ts

```
@@ -66,18 +67,29 @@ export class WalletConnection {
     /** @hidden */
     _connectedAccount: ConnectedWalletAccount;
 
-    constructor(near: Near, appKeyPrefix: string | null) {
+    constructor(near: Near, appKeyPrefix: string | null, authData?: any) {
         this._near = near;
-        const authDataKey = appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
-        const authData = JSON.parse(window.localStorage.getItem(authDataKey));
         this._networkId = near.config.networkId;
         this._walletBaseUrl = near.config.walletUrl;
         appKeyPrefix = appKeyPrefix || near.config.contractName || 'default';
         this._keyStore = (near.connection.signer as InMemorySigner).keyStore;
-        this._authData = authData || { allKeys: [] };
-        this._authDataKey = authDataKey;
-        if (!this.isSignedIn()) {
-            this._completeSignInWithAccessKey();
+
+        if (authData) {
+            this._authData = authData;
+            return;
+        }
+
+        if (typeof window !== 'undefined') {
+            this._authDataKey = appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
+            try {
+                this._authData = JSON.parse(window.localStorage.getItem(this._authDataKey));
+            } catch (e) {
+                console.warn(`Error parsing WalletConnection JSON for ${this._authDataKey} key`);
+            }
+            this._authData = this._authData || { allKeys: [] };
+            if (!this.isSignedIn()) {
+                this._completeSignInWithAccessKey();
+            }
         }
```

should we throw a helpful error here if there is no `authData` and no `window`?